cpp/tests/FloeE2ETests.cpp

```diff
@@ -0,0 +1,139 @@
+#include <catch2/catch_test_macros.hpp>
+#include "TestUtils.hpp"
+#include <random>
+
+using floe::test::createTestKey;
+
+TEST_CASE("E2E encryption and decryption with various plaintext sizes", "[e2e]") {
+    auto key = createTestKey();
+    auto parameterSpec = floe::FloeParameterSpec::GCM256_SHA384_4K;
+
+    auto floe = std::make_unique<floe::Floe>(parameterSpec);
+
+    std::vector testSizes = {
+        100,
+        parameterSpec.getPlainTextSegmentLength() - 1,
+        parameterSpec.getPlainTextSegmentLength(),
+        parameterSpec.getPlainTextSegmentLength() + 1,
+        parameterSpec.getPlainTextSegmentLength() * 2,
+        parameterSpec.getPlainTextSegmentLength() * 2 + 1
+    };
+    
+    for (int plaintextSize : testSizes) {
+        size_t aadLength = 0;
+        const uint8_t* aad = nullptr;
+        std::vector<uint8_t> plaintext(plaintextSize);
+        std::random_device rd;
+        std::mt19937 gen(rd());
+        std::uniform_int_distribution dis(0, 255);
+        
+        for (auto& byte : plaintext) {
+            byte = static_cast<uint8_t>(dis(gen));
+        }
```

Comment:
This is way too complicated. Just use `getentropy` from `sys/random.h`
```suggestion
        getentropy(plaintext.data(), plaintext.size());
```